transformer/app.js

```diff
@@ -101,21 +118,122 @@ function transformIssue(body, type, accountId) {
   });
 }
 
+const generateUpdatePayload = (existingFindingsToUpdate) => existingFindingsToUpdate.map((finding) => ({
+  FindingIdentifiers: [{
+    Id: finding.Id,
+    ProductArn: finding.ProductArn,
+  }],
+  Severity: {
+    Label: finding.FindingProviderFields.Severity.Label,
+  },
+}));
+
+const writeParams = (id, timestamp) => ({
+  TableName: 'xray-findings',
+  Item: {
+    ID: id,
+    TIMESTAMP: timestamp,
+  },
+});
+
+const writeFindingsToDB = async (findingsCollection) => {
+  const promises = [];
+  findingsCollection.map(async (finding) => {
+    const id = finding.Id;
+    const timestamp = new Date().getTime().toString();
+    try {
+      promises.push(ddbClient.send(new PutCommand(writeParams(id, timestamp))));
+      console.log(`Writing ID to DB: ${finding.Id}`);
+    } catch (err) {
+      console.error(err);
+    }
+  });
+  const results = await Promise.allSettled(promises);
+  const resultsFulfilled = results.filter((result) => (result.status === 'fulfilled')).map((result) => result.status);
+  const resultsFailed = results.filter((result) => (result.status !== 'fulfilled')).map((result) => result.message);
+  return {
+    SuccessCount: resultsFulfilled.length,
+    SuccessfulRecords: resultsFulfilled,
+    FailedRecords: resultsFailed.length,
+    FailedCount: resultsFailed,
+  };
+};
+
+const queryParams = (id) => ({
+  TableName: 'xray-findings',
+  ProjectionExpression: 'ID',
+  KeyConditionExpression: 'ID = :id',
+  Limit: 1,
+  ExpressionAttributeValues: {
+    ':id': { S: id },
+  },
+});
+
+const verifyFindingIds = async (ids) => {
+  const promises = ids.map((id) => ddbDocClient.send(new QueryCommand(queryParams(id))));
+  const results = await Promise.allSettled(promises);
+  return results.filter((result) => (result.status === 'fulfilled' && result.value.Count > 0)).map((result) => {
+    console.debug(JSON.stringify(result.value.Items[0].ID.S));
+    return result.value.Items[0].ID.S;
+  });
+};
+
+const sendUpdateCommand = async (updateBody) => {
+  const promises = updateBody.map((body) => hubClient.send(new BatchUpdateFindingsCommand(body)));
+  const results = await Promise.allSettled(promises);
+  console.debug(`Unfiltered results (promises): ${JSON.stringify(results)}`);
+  const resultsFulfilled = results.filter((result) => (result.status === 'fulfilled')).map((result) => result.value.ProcessedFindings);
+  const resultsFailed = results.filter((result) => (result.status !== 'fulfilled')).map((result) => result.value.ProcessedFindings);
```

Comment:
Let's use `result.status === 'rejected'` instead to be clear.

Reply:
Ok